routing/http/types/iter/filter.go

```diff
@@ -0,0 +1,43 @@
+package iter
+
+// Filter returns an iterator that filters out values that don't satisfy the predicate f.
+func Filter[T any](iter Iter[T], f func(t T) bool) *FilterIter[T] {
+	return &FilterIter[T]{iter: iter, f: f}
+}
+
+type FilterIter[T any] struct {
+	iter Iter[T]
+	f    func(T) bool
+
+	done bool
+	val  T
+}
```

Comment:
NOTE: It would be nicer to use [go iterators](https://pkg.go.dev/iter) here to allow `range` to be used to iterate the filters. However, that requires go1.23 . If only we were requiring that.